influxdb_influxql_parser/src/expression/conditional.rs

```diff
@@ -185,6 +197,16 @@ pub(crate) fn conditional_expression(i: &str) -> ParseResult<&str, ConditionalEx
     disjunction(i)
 }
 
+impl FromStr for ConditionalExpression {
+    type Err = ();
```

Comment:
I wonder why not return the underlying error 

```suggestion
    type Err = &str;
```

And then I think you could avoid the `map_err` below 🤔

Reply:
I'll do better still and return a `ParseError`, consistent with the `parse_statements` public function. Thanks for pointing that one out!

Reply:
Better yet, I've introduced a `parse_conditional_expression` public API that, similar to the `parse_statements`, consumes the entire input to ensure the input is a valid `ConditionalExpression`. Like the `parse_statements` function, `parse_conditional_expression` returns a `ParseError` for any failures. The `parse_conditional_expression` function is used to implement the `FromStr` trait.